low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +40,145 @@ class ServerMap(ResourceMap):
 
     def __init__(self):
         """Initialize server"""
-        pass
+        super().__init__()
+        self.sysfs = ToolFactory().get_instance('sysfs')
+        self.sysfs.initialize()
+        self.sysfs_base_path = self.sysfs.get_sysfs_base_path()
+        self.cpu_path = self.sysfs_base_path + CPU_PATH
+        self.server_frus = {
+            'cpu': self.get_cpu_info
+        }
 
     def get_health_info(self, rpath):
         """
         Get health information of fru in given resource id
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        info = {}
+        fru_found = False
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            for fru in self.server_frus:
+                info.update({fru: self.server_frus[fru]()})
+            info["last_updated"] = int(time.time())
+            fru_found = True
+        else:
+            fru = None
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_frus.get(fru):
+                    fru_found = True
+                    info = self.server_frus[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                f"Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    @staticmethod
+    def get_node_details(node):
```

Comment:
refer https://github.com/Seagate/cortx-monitor/pull/416/files
this function can stay in resource_map.py itself, lets follow that

Reply:
@cdeshmukh Done.